allensdk/brain_observatory/behavior/project_apis/data_io/behavior_project_cloud_api.py

```diff
@@ -316,3 +345,17 @@ def get_natural_scene_template(self, number: int) -> Iterable[bytes]:
         :returns: iterable yielding a tiff file as bytes
         """
         raise NotImplementedError()
+
+    def _get_metadata_path(self, fname: str):
+        if self._local:
+            path = self.cache.metadata_path(fname=fname)['local_path']
```

Comment:
The `cache.metadata_path` method returns a dict with element `exists` which checks

a) that the file exists
b) that it's has aligns with what the manifest says to expect

If `cache.metadata_path()['exists']` comes back `False`, we should alert the user and not load any data since there either is no data or the data has become corrupted.

Technically `cache.download_metadata` will not do any downloading if the file already exists in a valid state, so we could alternatively add an error to `_download_file` that alerts the user when they are offline and rely on `download_metadata` to do the right thing when they have already downloaded the metadata.